Telegram/SourceFiles/boxes/sticker_set_box.cpp

```diff
@@ -198,6 +198,17 @@ void StickerSetBox::copyStickersLink() {
 	QGuiApplication::clipboard()->setText(url);
 }
 
+void StickerSetBox::archiveStickers() {
+	_controller->session().api().request(MTPmessages_InstallStickerSet(
+		_set,
+		MTP_boolTrue()
+	)).done([=](const MTPmessages_StickerSetInstallResult &result) {
+		Ui::Toast::Show(tr::lng_stickers_has_been_archived(tr::now));
+	}).fail([=](const MTP::Error &error) {
```

Comment:
I'm afraid it's not so simple.
You have to apply the current freshly archived sticker set locally and remove it from the list of installed sticker sets.
I think that would be enough for now.

```suggestion
	const auto weak = base::make_weak(_controller.get());
	const auto setId = _set.c_inputStickerSetID().vid().v;
	_controller->session().api().request(MTPmessages_InstallStickerSet(
		_set,
		MTP_boolTrue()
	)).done([=](const MTPmessages_StickerSetInstallResult &result) {
		const auto controller = weak.get();
		if (!controller) {
			return;
		}
		if (result.type() == mtpc_messages_stickerSetInstallResultSuccess) {
			Ui::Toast::Show(tr::lng_stickers_has_been_archived(tr::now));

			const auto &session = controller->session();
			auto &order = session.data().stickers().setsOrderRef();
			const auto index = order.indexOf(setId);
			if (index == -1) {
				return;
			}
			order.removeAt(index);

			session.local().writeInstalledStickers();
			session.local().writeArchivedStickers();

			session.data().stickers().notifyUpdated();
		}
	}).fail([](const MTP::Error &error) {
```